### ISSUE TITLE
Problems with gfortran-4.8 when updating R packages

### ISSUE THREAD

**Issue by SplashDance** [Homebrew/homebrew#43290]
When upgrading **R** packages which need compilation I've gotten errors such as:

    make: gfortran-4.8: No such file or directory

and:

    ld: warning: directory not found for option '-L/usr/local/lib/gcc/x86_64-apple-darwin13.0.0/4.8.2'
    ld: library not found for -lgfortran

This is a new issue for me--in the past I've compiled **R** package updates with no problems--and it at least *sounds* like this problem results from having upgraded my brew-installed version of **gcc** to version 5.2 (in addition, from [this issue](https://github.com/Homebrew/homebrew/pull/28855) it sounds like it could be related to the **gfortran** suffix being omitted, or something like that).

So my question is this: Is there a way I can fix this problem using my currently installed version of **gcc** or do I need to reinstall **gfortran-4.8.2**? [1]

Thanks for your patience,
Steve

[1] I mention this because several resources (such as [this one](http://stackoverflow.com/a/25912049/3309177)) suggest I should run this code in the terminal:

    curl -O http://r.research.att.com/libs/gfortran-4.8.2-darwin13.tar.bz2
    sudo tar fvxz gfortran-4.8.2-darwin13.tar.bz2 -C /

Would running this code end up causing a conflict with my already-existing version of **gcc**?

**Comment by mikemcquaid** [comment]
How did you install R?

**Comment by tdsmith** [comment]
Please run `brew reinstall r` and please do not install the r.research.att.com compilers.

This was fixed in https://github.com/Homebrew/homebrew-science/commit/880a56712005fd7a80a29236fbbe71d63a219d96. I should push a revision bump since others will run into this.

**Comment by tdsmith** [comment]
Oops, BrewTestBot bottle commits close cross-tap issues.

**Comment by SplashDance** [comment]
One thing, though: I originally installed **R** version 3.2.2 via [CRAN's download page](https://cran.r-project.org/bin/macosx/), *not* Homebrew. Is that going to be a problem?

Should I uninstall that package (via [these directions](https://cran.r-project.org/doc/manuals/r-release/R-admin.html#Uninstalling-under-OS-X)) and then *re*-install **R** using Homebrew?

**Comment by tdsmith** [comment]
This should work if you do that. I don't know how the CRAN builds are intended to discover host compilers.

**Comment by tdsmith** [comment]
Building packages should succeed if you uninstall the CRAN R and install the Homebrew R distribution. I don't know how the CRAN builds are intended to discover host compilers; it would be good to look into that.

**Comment by SplashDance** [comment]
Ok, I uninstalled **R** and then re-installed it via Homebrew and have been re-installing the packages I ended up deleting.

Now I just have a couple of quick questions:
* Where should I be installing my R packages? So far they are being installed to `/usr/local/lib/R/3.2/site-library`. Should I instead be installing them to `/usr/local/Cellar/r/3.2.2_1/R.framework/Versions/3.2/Resources/library/`?
* Will upgrading **R** via Homebrew (say, to R 3.3.0 when released) force me to re-install all the packages I've already installed?
* The only problem I've come across when re-installing my **R** packages is the error:

        checking for GTK... no
        configure: error: GTK version 2.8.0 required
        ERROR: configuration failed for package ‘RGtk2’

    Does this mean I should install **gtk** via homebrew? (And, if so, would it be "**gtk+**" or "**gtk+3**"?)

Thanks again for your help, guys!

**Comment by tdsmith** [comment]
To answer both questions at once, installing to /usr/local/lib/R/3.2/site-library is a good idea because packages installed to that library will remain visible across patch-level upgrades to R (for example, to 3.2.3 or 3.2.2_2) but not across minor-version upgrades (like to 3.3.0). This should be the optimum behavior because I believe that R does not promise that compiled extensions remain compatible across minor-version updates (i.e. 3.2 to 3.3).

gtk version 2.x is provided by the gtk+ formula.

**Comment by SplashDance** [comment]
Ok, terrific--you've been really helpful.

As to the **gtk+** question, I've installed it but now I'm getting an error when trying to install the **Rgtk** package:

    Rgtk.c:7:10: fatal error: 'gdk/gdkx.h' file not found
    #include <gdk/gdkx.h>
             ^

Know anything about it?

If not, that's ok--I've tried googling for a solution to no avail but I'm sure I'll figure out a workaround eventually. Thanks again for all your help.

-Steve

**Comment by tdsmith** [comment]
Ah; Homebrew's gtk+ doesn't provide gdkx.h because our gtk+ targets Quartz and not X11. Did you have to build this from source with a CRAN R?

**Comment by SplashDance** [comment]
I don't recall ever having to install **gtk+** for a CRAN **R** but I believe I did need to first download **XQuartz** before installing one of CRAN's **R**s. Or maybe I installed **XQuartz** when first trying to install **Octave**--either way, I did (and still do) have **XQuartz** installed on my system.

**Comment by tdsmith** [comment]
You may be able to pass `type="binary"` to `install.packages` to get a binary download from CRAN but I don't know if it will work.

**Comment by SplashDance** [comment]
Ok, I tried passing in the argument `type='mac.binary.mavericks'` and things installed but then when I ran **rattle** it asked me if I wanted to install **GTK+** and then an installer for **GTK+ 2.24.17 (X11)** popped up. And that's where I am now--I just sort of left the installer waiting for me to click 'Continue' since I wasn't sure if installing it would be a bad idea or not. Any advice on how to proceed?
